src/components/PublicGroupList.vue

```diff
@@ -154,6 +159,33 @@ export default {
             }
             return monitor.element.sendUrl && monitor.element.url && monitor.element.url !== "https://" && !this.editMode;
         },
+
+        /**
+         * Returns formatted certificate expiry or Bad cert message
+         * @param {Object} monitor Monitor to show expiry for
+         * @returns {string}
+         */
+        formattedCertExpiryMessage(monitor) {
+            if (monitor?.element?.validCert && monitor?.element?.certExpiryDaysRemaining) {
+                return monitor.element.certExpiryDaysRemaining + " " + this.$tc("day", monitor.element.certExpiryDaysRemaining);
+            } else if (monitor?.element?.validCert === false) {
+                return this.$t("noOrBadCertificate");
+            } else {
+                return "";
+            }
+        },
+
+        /**
+         * Returns certificate expiry based on days remaining
+         * @param {Object} monitor Monitor to show expiry for
+         * @returns {string}
+         */
+        certExpiryColor(monitor) {
+            if (monitor?.element?.validCert) {
+                return monitor.element.certExpiryDaysRemaining > 7 ? "primary" : "danger";
+            }
+            return "danger";
```

Comment:
I think the following is more readable (unshure if nullisch coalessing is needed as well).
```suggestion
            if (monitor?.element?.validCert && monitor.element.certExpiryDaysRemaining > 7) {
                return "primary";
            }
            return "danger";
```

Reply:
Fixed